openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/slack/SlackApp.java

```diff
@@ -0,0 +1,287 @@
+        .clientId(config.getClientId())
+        .clientSecret(config.getClientSecret())
+        .signingSecret(config.getSigningCertificate())
+        .scope(config.getScopes())
+        .oauthInstallPath(SLACK_OAUTH_INSTALL_ENDPOINT)
+        .oauthRedirectUriPath(SLACK_OAUTH_CALLBACK_ENDPOINT)
+        .stateValidationEnabled(true)
+        .build();
+  }
+
+  public String buildOAuthUrl() {
+    String clientId = appConfig.getClientId();
+    String scopes = appConfig.getScopes();
+    String callbackUrl = appConfig.getCallbackUrl();
+
+    String state = generateRandomState();
+    saveTokenToSystemRepository(state, SettingsType.SLACK_O_AUTH_STATE);
+
+    return SLACK_OAUTH_BASE_URL
+        + "?client_id="
+        + clientId
+        + "&scope="
+        + scopes
+        + "&state="
+        + state
+        + "&redirect_uri="
+        + callbackUrl;
+  }
+
+  @Override
+  public void raisePreviewMessage(org.openmetadata.schema.entity.app.App app) {
+    throw AppException.byMessage(
+        app.getName(), "Preview", "Contact Collate to purchase the Application");
+  }
+
+  public String getRedirectUrl(boolean isStateValid) {
+    return appConfig.getCallbackRedirectURL() + (isStateValid ? SUCCESS_FRAGMENT : FAILED_FRAGMENT);
+  }
+
+  public String saveTokenAndBuildRedirectUrl(String code) {
+    boolean isSuccess = exchangeCodeAndSaveToken(code);
+    return appConfig.getCallbackRedirectURL() + (isSuccess ? SUCCESS_FRAGMENT : FAILED_FRAGMENT);
+  }
+
+  private boolean exchangeCodeAndSaveToken(String code) {
+    OkHttpClient client = new OkHttpClient();
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    try {
+      RequestBody body = buildTokenExchangeRequestBody(code);
+      Request request = new Request.Builder().url(SLACK_TOKEN_EXCHANGE_ENDPOINT).post(body).build();
+
+      try (okhttp3.Response slackResponse = client.newCall(request).execute()) {
+        if (!slackResponse.isSuccessful()) {
+          throw new IOException("Unexpected code " + slackResponse.code());
+        }
+
+        String responseBody = slackResponse.body().string();
+        JsonNode jsonNode = objectMapper.readTree(responseBody);
+        String botAccessToken = extractAccessToken(jsonNode, "access_token", "Bot access token");
+        String userAccessToken =
+            extractAccessToken(
+                jsonNode.get("authed_user"), "access_token", "Authed user access token");
+
+        saveTokenToSystemRepository(botAccessToken, SettingsType.SLACK_BOT);
+        saveTokenToSystemRepository(userAccessToken, SettingsType.SLACK_INSTALLER);
+
+        return true;
+      } catch (IOException e) {
+        LOG.error("Error executing Slack token exchange request: {}", e.getMessage());
+        return false;
+      }
+    } catch (Exception e) {
+      LOG.error("Error exchanging Slack token: {}", e.getMessage());
+      return false;
+    }
+  }
+
+  private RequestBody buildTokenExchangeRequestBody(String code) {
```

Comment:
we don't need to these manually bolt-idk can help us here

Reply:
When using bolt-servlet, it needs to live with the server. The callback web servlet has to be loaded when the server starts with the initialized SlackApp. Otherwise, if we start the SlackApp, those servlets need to be dynamically loaded on the server.

Reply:
Removed buildTokenExchangeRequestBody method and similar, now using slack-api-methods for the token exchange.